R/object.R

```diff
@@ -36,7 +36,7 @@ new_object <- function(.data = NULL, ...) {
   }
 
   attr(object, "object_class") <- obj_cls
-  class(object) <- class_dispatch(obj_cls)
+  class(object) <- setdiff(class_dispatch(obj_cls), "ANY")
```

Comment:
I see you had to do this exact pattern at least twice. Mainly just pointing that out, not sure if a different named helper is needed or not